File: src/agents/tools/sources/ynet.py
import logging

import feedparser

from agents.tools.sources.common.rss import RSSFeed

logger = logging.getLogger(__name__)


def get_ynet() -> RSSFeed:
	"""
	Fetch slightly left-leaning news outlet YNet.
	Political orientation: -3 (On a scale of [-10, -10])
	"""

	raw_feed = feedparser.parse("https://www.ynet.co.il/Integration/StoryRss2.xml")
	logger.debug(f"Fetched the news from YNet {raw_feed=}")

	return RSSFeed.model_validate(raw_feed)

File: src/agents/tools/sources/ynet.ts
import Parser from "rss-parser";
import type { RSSFeed } from "./common/rss.ts";

const parser = new Parser();

export async function getYnet(): Promise<RSSFeed> {
	const feed = await parser.parseURL(
		"https://www.ynet.co.il/Integration/StoryRss2.xml",
	);

	return {
		entries: feed.items.map((item) => ({
			title: item.title || "",
			link: item.link || "",
			summary: item.contentSnippet || item.content || "",
			published: item.pubDate || "",
		})),
	};
}
